packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -44,6 +44,9 @@ export function registerComponent(
     Ctor: ComponentConstructor,
     { name, tmpl: template }: { name: string; tmpl: Template }
```

Comment:
Based on the types, `name` can't be undefined. The type definition needs to be realigned.